src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs

```diff
@@ -317,6 +323,39 @@ public async void Execute()
             }
         }
 
+        public void Initialize()
+        {
+            if (_initialized)
+            {
+                return;
+            }
+
+            _initialized = true;
+
+            Request = new Request(this);
+            Response = new Response(this);
+
+            _features = new FeatureCollection(new StandardFeatureCollection(this));
+            _enableResponseCaching = Server.Options.EnableResponseCaching;
+
+            // Pre-initialize any fields that are not lazy at the lower level.
+            _requestHeaders = Request.Headers;
+            _httpMethod = Request.Method;
+            _path = Request.Path;
+            _pathBase = Request.PathBase;
+            _query = Request.QueryString;
+            _rawTarget = Request.RawUrl;
+            _scheme = Request.Scheme;
+
+            if (Server.Options.Authentication.AutomaticAuthentication)
+            {
+                _user = User;
+            }
+
+            _responseStream = new ResponseStream(Response.Body, OnResponseStart);
+            _responseHeaders = Response.Headers;
```

Comment:
Can you move this part back to an InitializeFeatures method in RequestContext.FeatureCollection where most of these fields are declared? Maintenance wise I'm concerned about people understanding that all the fields in RequestContext.FeatureCollection are intentional duplicates so that the application can mutate them without affecting the server functionality.

Reply:
Yep

Reply:
Done